gunrock/applications/sssp/sssp_implementation.hxx

```diff
@@ -11,165 +11,131 @@
  */
 #pragma once
 
-#include <gunrock/applications/application.hxx>
 #include <bits/stdc++.h>
 
+#include <gunrock/applications/application.hxx>
+
 namespace gunrock {
 namespace sssp {
 
 template <typename meta_t>
-struct sssp_param_t {
-   using vertex_t = typename meta_t::vertex_type;
-   
-   vertex_t single_source;
+struct param_t {
+  using vertex_t = typename meta_t::vertex_type;
    
-   sssp_param_t(
-     vertex_t single_source_
-   ) {
-     single_source = single_source_;
-   }
+  vertex_t single_source;
+  param_t(
+    vertex_t single_source_
+  ) :
+    single_source(single_source_) 
+  {}
 };
 
 template <typename meta_t>
-struct sssp_result_t {
+struct result_t {
   using vertex_t = typename meta_t::vertex_type;
   using weight_t = typename meta_t::weight_type;
    
   weight_t* distances;
   vertex_t* predecessors;
   
-  sssp_result_t(
+  result_t(
     weight_t* distances_,
     vertex_t* predecessors_
-  ) {
-    distances    = distances_;
-    predecessors = predecessors_;
-  }
+  ) :
+    distances(distances_), 
+    predecessors(predecessors_) 
+  {}
 };
 
-
-template <typename d_graph_t, typename meta_t>
-struct sssp_problem_t : problem_t<d_graph_t, meta_t> {
-
-  using param_t   = sssp_param_t<meta_t>;
-  using result_t  = sssp_result_t<meta_t>;
-
+template <typename graph_t, typename meta_t, typename param_t, typename result_t>
+struct problem_t : gunrock::problem_t<graph_t, meta_t, param_t, result_t> {
+  // Use Base class constructor -- does this work? does it handle copy constructor?
+  using gunrock::problem_t<graph_t, meta_t, param_t, result_t>::problem_t;
```

Comment:
I think this is making base class constructor accessible as public within `sssp::problem_t`, Idk what you're trying to achieve with this.

Reply:
This makes it so the user doesn't have to include the boilerplate constructor code in this file.  All of the things that happen in the body of the constructor get moved to `init` and `reset`, which I like for this and other reasons, but open to changing back.